src/Products/ProductHydrator.php

```diff
@@ -13,4 +14,22 @@ public static function getProductsByCategory($categoryID)
         $query->setFetchMode(PDO::FETCH_CLASS, SimpleProductEntity::class);
         return $query->fetchALL();
     }
+
+    public static function getProductById($productId)
+    {
+        $db = DbConnection::setConnection();
+        $query = $db -> prepare ('SELECT `id`,`name`,`category_id`,`width`,`height`,`depth`,`price`,`stock`,`related`,`color` FROM `products` WHERE `id`= :productId');
+        $query->execute(['productId'=>$productId]);
+        $query->setFetchMode(PDO::FETCH_CLASS, DetailedProductEntity::class);
+        return $query->fetch();
+    }
+    public static function getMaxProducts()
+    {
+        $db = DbConnection::setConnection();
```

Comment:
Rename the method in `DbConnection` to `getConnection`, as it's getting an existing connection

Reply:
(even though it sets it the first time!)